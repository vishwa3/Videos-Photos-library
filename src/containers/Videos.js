import React, { /* useEffect, */ useRef } from "react";
/* import { useDispatch, useSelector } from "react-redux"; */
import { useDispatch, useSelector } from './react-redux-hooks';
import { searchMediaAction } from "../actions/mediaActions";
import "../styles/style.css";
import Loader from "./Loader";

function Videos() {
  const query = useRef(null);
  const dispatch = useDispatch();

  React.useEffect(() => {
    dispatch(searchMediaAction("dakota", "video"));
  }, []);

  const videos = useSelector(state => state.vid.videoArray);
  const error = useSelector(state => state.errorReducer.error);

  const handleSearch = event => {
    console.log('hello world'); // console statement for the purpose of JEST/Enzyme testing
    event.preventDefault();
    if (query.current.value !== null && query.current.value !== "") {
      dispatch(searchMediaAction(query.current.value, "video"));
      query.current.value = "";
    }
  };
  const handleKeyPress = event => {
    if (event.key === 'Enter' && query.current.value !== null && query.current.value !== "") {
      dispatch(searchMediaAction(query.current.value, "video"));
      query.current.value = "";
    }
  }

  return (
    <div className="container-fluid">
      {!error ? (<div>{videos.length > 0 ? (
        <div>
          <div className="searchButton">
            <input
              type="text"
              onKeyPress={handleKeyPress}
              style={{ marginRight: 5 }}
              ref={query}
            />
            <input
              type="submit"
              className="btn btn-primary"
              value="Search Videos"
              onClick={handleSearch}
            />
          </div>
          <div>
            <h2 className="heading">Videos</h2>
            <div style={{ textAlign: "center" }}>
              <section className="videogrid">
                {videos.map(video => (
                  <div key={video.id}>
                    <video controls src={video.mediaUrl} alt={video.description} />
                  </div>
                ))}
              </section>
            </div>
          </div>
        </div>
      ) : (<Loader />)}</div>) : (<div><div className="searchButton">
        <input
          type="text"
          onKeyPress={handleKeyPress}
          style={{ marginRight: 5 }}
          ref={query}
        />
        <input
          type="submit"
          className="btn btn-primary"
          value="Search Videos"
          onClick={handleSearch}
        />
      </div><h3>Error</h3>{error && <div className="error">{JSON.stringify(error)}</div>}</div>)}

    </div>
  )

}

export default Videos;