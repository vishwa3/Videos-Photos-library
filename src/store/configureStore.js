import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import rootReducer from '../reducers/index';
import rootSaga from '../sagas'; // TODO: Next step
import thunk from "redux-thunk";

//  Returns the store instance
// It can  also take initialState argument when provided
const configureStore = () => {
  const sagaMiddleware = createSagaMiddleware(); 
  return {
    ...createStore(rootReducer,
      applyMiddleware(sagaMiddleware,thunk)),
    runSaga: sagaMiddleware.run(rootSaga)
  };
};

export default configureStore;