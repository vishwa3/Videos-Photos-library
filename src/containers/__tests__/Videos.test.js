import React from 'react';
import renderer from 'react-test-renderer';
import Videos from '../Videos';
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure,mount } from 'enzyme';
import * as ReactReduxHooks from '../react-redux-hooks'

configure({adapter: new Adapter()});

describe("Photos", () => {
  let wrapper, useEffect, store;

  const mockUseEffect = () => {
    useEffect.mockImplementationOnce(f => f());
  };


  beforeEach(() => {
    store = configureStore()({});
    useEffect = jest.spyOn(React, "useEffect");
    mockUseEffect();
    mockUseEffect();

    jest.spyOn(ReactReduxHooks, "useSelector").mockImplementation(state => store.getState());

    jest.spyOn(ReactReduxHooks, "useDispatch").mockImplementation(() => store.dispatch);

    wrapper = shallow(<Videos store={store} />);
  })

  describe("on start", () => {
    it("dispatch search Media Action to store", () => {
      const actions = store.getActions();
      expect(actions).toEqual([{ type: 'SEARCH_MEDIA_REQUEST', payload: { searchCategory: "video", searchValue: "dakota" } }])
    })

  })

  it("snapshot of Videos component", () => {
    const component = renderer.create(<Videos store={store} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should have search field', () => {
    expect(wrapper.find('input[type="text"]').length).toEqual(1);
  })

  it('should have search Videos button', () => {
    expect(wrapper.find('input[type="submit"]').length).toEqual(1)
  })

  it('testing click', () => {
    const logSpy = jest.spyOn(console, 'log');
    const container = mount(<Videos store={store} />);
    container.find('input[type="submit"]').simulate("click");
    expect(logSpy).toBeCalledWith('hello world')

  })
})
