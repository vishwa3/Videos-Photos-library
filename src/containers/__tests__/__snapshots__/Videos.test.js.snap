// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos snapshot of Videos component 1`] = `
<div
  className="container-fluid"
>
  <div>
    <div
      className="searchButton"
    >
      <input
        onKeyPress={[Function]}
        style={
          Object {
            "marginRight": 5,
          }
        }
        type="text"
      />
      <input
        className="btn btn-primary"
        onClick={[Function]}
        type="submit"
        value="Search Videos"
      />
    </div>
    <h3>
      Error
    </h3>
    <div
      className="error"
    >
      {}
    </div>
  </div>
</div>
`;
